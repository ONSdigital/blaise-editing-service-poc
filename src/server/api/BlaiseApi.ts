import BlaiseClient, {
  CaseData,
  CaseEditInformation, CaseResponse, Questionnaire, User,
} from 'blaise-api-node-client';
import { ServerConfiguration } from '../interfaces/serverConfigurationInterface';
import { QuestionnaireDetails } from '../../common/interfaces/surveyInterface';
import mapQuestionnaireDetails from '../mappers/questionnaireMapper';

export default class BlaiseApi {
  config: ServerConfiguration;

  blaiseApiClient: BlaiseClient;

  constructor(config: ServerConfiguration, blaiseApiClient: BlaiseClient) {
    this.config = config;
    this.blaiseApiClient = blaiseApiClient;
    this.getQuestionnaires = this.getQuestionnaires.bind(this);
    this.getCase = this.getCase.bind(this);
    this.getCaseEditInformation = this.getCaseEditInformation.bind(this);
    this.getUsers = this.getUsers.bind(this);
    this.updateCase = this.updateCase.bind(this);
  }

  async getQuestionnaires(): Promise<QuestionnaireDetails[]> {
    const questionnaires = await this.blaiseApiClient.getQuestionnaires(this.config.ServerPark);

    const questionnaireDetailsList: QuestionnaireDetails[] = [];
    questionnaires.forEach((questionnaire : Questionnaire) => {
      if (questionnaire.name.endsWith('_EDIT')) {
        questionnaireDetailsList.push(mapQuestionnaireDetails(questionnaire));
      }
    });

    return questionnaireDetailsList;
  }

  async getCase(questionnaireName: string, caseId: string): Promise<CaseResponse> {
    return this.blaiseApiClient.getCase(this.config.ServerPark, questionnaireName, caseId);
  }

  async updateCase(questionnaireName: string, caseId: string, caseFields: CaseData): Promise<void> {
    await this.blaiseApiClient.updateCase(this.config.ServerPark, questionnaireName, caseId, caseFields);
  }

  async getCaseEditInformation(questionnaireName: string): Promise<CaseEditInformation[]> {
    const caseEditInformationList = await this.blaiseApiClient.getCaseEditInformation(this.config.ServerPark, questionnaireName);

    caseEditInformationList.forEach((caseEditInformation) => {
      caseEditInformation.editUrl = `https://${this.config.ExternalWebUrl}/${questionnaireName}?KeyValue=${caseEditInformation.primaryKey}`;
    });

    return caseEditInformationList;
  }

  async getUsers(): Promise<User[]> {
    return this.blaiseApiClient.getUsers();
  }
}
