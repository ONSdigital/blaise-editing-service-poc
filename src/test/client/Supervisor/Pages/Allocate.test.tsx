import {
  render, act, RenderResult, fireEvent,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Router from 'react-router';
import { getAllocationDetails, updateAllocationDetails } from '../../../../client/api/NodeApi';
import UserRole from '../../../../client/Common/enums/UserTypes';
import { AllocationDetails } from '../../../../common/interfaces/allocationInterface';
import AllocationMockObject from '../../MockObjects/AllocationMockObjects';
import Allocate from '../../../../client/Supervisor/Pages/Allocate';

// set global vars
const supervisorRole:UserRole = UserRole.SVT_Supervisor;
const editorRole:UserRole = UserRole.SVT_Editor;
const questionnaireName = 'FRS2504A';
let view:RenderResult;

// set mocks
/* eslint import/no-extraneous-dependencies: 0 */
jest.mock('react-router', () => ({ ...jest.requireActual('react-router'), useParams: jest.fn() }));
jest.spyOn(Router, 'useParams').mockReturnValue({ questionnaireName });

jest.mock('../../../../client/api/NodeApi');
const getAllocationDetailsMock = getAllocationDetails as jest.Mock<Promise<AllocationDetails>>;
const updateAllocationDetailsMock = updateAllocationDetails as jest.Mock<Promise<void>>;

describe('Given we wish to allocte cases from an Interviewer to an Editor', () => {
  const reallocate = false;

  beforeEach(() => {
    getAllocationDetailsMock.mockReturnValue(Promise.resolve(AllocationMockObject));
    updateAllocationDetailsMock.mockResolvedValue();
  });

  afterEach(() => {
    getAllocationDetailsMock.mockReset();
    updateAllocationDetailsMock.mockReset();
  });

  it('should render the allocation page correctly', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });

  it('should display an have the correct page info for allocation', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const panelView = view.getByTestId('allocation-page-panel');
    expect(panelView).toHaveTextContent('Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor');
  });

  it('should display alist of availiable interviewer for alloction from', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const interviewerListOption = view.getByTestId('select-from');
    expect(interviewerListOption.childElementCount).toEqual(AllocationMockObject.Interviewers.length + 1);

    expect(interviewerListOption.children[0]).toHaveTextContent('Select an option');
    AllocationMockObject.Interviewers.forEach((interviewer, interviewerIndex) => {
      expect(interviewerListOption.children[interviewerIndex + 1]).toHaveTextContent(`${interviewer.Name} (${interviewer.Cases.length} case(s))`);
    });
  });

  it('should display alist of availiable editors for alloction to', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const editorListOption = view.getByTestId('select-to');
    expect(editorListOption.childElementCount).toEqual(AllocationMockObject.Editors.length + 1);

    expect(editorListOption.children[0]).toHaveTextContent('Select an option');
    AllocationMockObject.Editors.forEach((Editor, editorIndex) => {
      expect(editorListOption.children[editorIndex + 1]).toHaveTextContent(`${Editor.Name} (${Editor.Cases.length} case(s))`);
    });
  });

  it('should call updateAllocationDetails with the expected parameters when the allocation button is clicked', async () => {
    // arrange
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'jamester' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Jake' } });
      fireEvent.click(view.getByText('Allocate'));
    });

    // assert
    expect(updateAllocationDetailsMock).toBeCalledWith(questionnaireName, 'Jake', ['10001013']);
  });

  it('should call updateAllocationDetails with the expected parameters when the allocation button is clicked is clicked and number of cases is limited to 1', async () => {
    // arrange
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'bob' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Jake' } });
      fireEvent.change(view.getByTestId('number-of-cases'), { target: { value: '1' } });
      fireEvent.click(view.getByText('Allocate'));
    });

    // assert
    expect(updateAllocationDetailsMock).toBeCalledWith(questionnaireName, 'Jake', ['10001011']);
  });

  it('should show a success message when allocation is successful', async () => {
    // arrange
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'jamester' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Jake' } });
      fireEvent.click(view.getByText('Allocate'));
    });

    // assert
    const successMessage = view.getByTestId('SuccessMessage');
    expect(successMessage).toHaveTextContent(`Case(s) '10001013' have been allocated to 'Jake' for '${questionnaireName}'`);

    expect(view.queryByTestId('ErrorMessage')).not.toBeInTheDocument();
  });

  it('should show an error message when allocation is not successful', async () => {
    // arrange
    updateAllocationDetailsMock.mockRejectedValue(new Error('Could not allocate cases'));

    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'jamester' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Jake' } });
      fireEvent.click(view.getByText('Allocate'));
    });

    // assert
    const errorMessage = view.getByTestId('ErrorMessage');
    expect(errorMessage).toHaveTextContent('Case(s) could not be allocated, please try again in a few seconds');

    expect(view.queryByTestId('SuccessMessage')).not.toBeInTheDocument();
  });
});

describe('Given we wish to reallocte cases from an Editor to another Editor', () => {
  const reallocate = true;

  beforeEach(() => {
    getAllocationDetailsMock.mockReturnValue(Promise.resolve(AllocationMockObject));
    updateAllocationDetailsMock.mockResolvedValue();
  });

  afterEach(() => {
    getAllocationDetailsMock.mockReset();
    updateAllocationDetailsMock.mockReset();
  });

  it('should render the allocation page correctly', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });

  it('should display an have the correct page info for reallocation', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const panelView = view.getByTestId('allocation-page-panel');
    expect(panelView).toHaveTextContent('Reallocate cases from one editor to another editor. All non-completed cases will be transfered');
  });

  it('should display a list of editors for realloction from', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const editorListOption = view.getByTestId('select-from');
    expect(editorListOption.childElementCount).toEqual(AllocationMockObject.Editors.length + 1);

    expect(editorListOption.children[0]).toHaveTextContent('Select an option');
    AllocationMockObject.Editors.forEach((editor, editorIndex) => {
      expect(editorListOption.children[editorIndex + 1]).toHaveTextContent(`${editor.Name} (${editor.Cases.length} case(s))`);
    });
  });

  it('should display a list of availiable editors for realloction to', async () => {
    // arrange && act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const editorListOption = view.getByTestId('select-to');
    expect(editorListOption.childElementCount).toEqual(AllocationMockObject.Editors.length + 1);

    expect(editorListOption.children[0]).toHaveTextContent('Select an option');
    AllocationMockObject.Editors.forEach((Editor, editorIndex) => {
      expect(editorListOption.children[editorIndex + 1]).toHaveTextContent(`${Editor.Name} (${Editor.Cases.length} case(s))`);
    });
  });

  it('should call updateAllocationDetails with the expected parameters when the allocation button is clicked', async () => {
    // arrange
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'Jake' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Rich' } });
      fireEvent.click(view.getByText('Reallocate'));
    });

    // assert
    expect(updateAllocationDetailsMock).toBeCalledWith(questionnaireName, 'Rich', ['10001012', '10001015']);
  });

  it('should call updateAllocationDetails with the expected parameters when the allocation button is clicked and number of cases is limited to 1', async () => {
    // arrange
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'Jake' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Rich' } });
      fireEvent.change(view.getByTestId('number-of-cases'), { target: { value: '1' } });
      fireEvent.click(view.getByText('Reallocate'));
    });

    // assert
    expect(updateAllocationDetailsMock).toBeCalledWith(questionnaireName, 'Rich', ['10001012']);
  });

  it('should show a success message when reallocation is successful', async () => {
    // arrange
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'Jake' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Rich' } });
      fireEvent.click(view.getByText('Reallocate'));
    });

    // assert
    const successMessage = view.getByTestId('SuccessMessage');
    expect(successMessage).toHaveTextContent(`Case(s) '10001012, 10001015' have been allocated to 'Rich' for '${questionnaireName}'`);

    expect(view.queryByTestId('ErrorMessage')).not.toBeInTheDocument();
  });

  it('should show an error message when allocation is not successful', async () => {
    // arrange
    updateAllocationDetailsMock.mockRejectedValue(new Error('Could not allocate cases'));

    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // act
    await act(async () => {
      fireEvent.change(view.getByTestId('select-from'), { target: { value: 'Jake' } });
      fireEvent.change(view.getByTestId('select-to'), { target: { value: 'Rich' } });
      fireEvent.click(view.getByText('Reallocate'));
    });

    // assert
    const errorMessage = view.getByTestId('ErrorMessage');
    expect(errorMessage).toHaveTextContent('Case(s) could not be allocated, please try again in a few seconds');

    expect(view.queryByTestId('SuccessMessage')).not.toBeInTheDocument();
  });
});
