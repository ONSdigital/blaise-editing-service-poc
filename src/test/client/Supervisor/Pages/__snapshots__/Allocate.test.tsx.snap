// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given we wish to allocte cases from an Interviewer to an Editor should render the allocation page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        data-testid="allocation-page-panel"
      >
        <div
          class="ons-panel__body "
        >
          Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor
        </div>
      </div>
      <br />
      <h3>
        FRS2504A
      </h3>
      <div>
        <label
          class="ons-label"
          for="select-from"
        >
          Allocate cases from interviewer
           
        </label>
        <select
          class="ons-input "
          data-testid="select-from"
          id="select-from"
          name="select"
        >
          <option
            data-testid="select-select-from"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-from-bob"
            value="bob"
          >
            bob (2 case(s))
          </option>
          <option
            data-testid="option-select-from-jamester"
            value="jamester"
          >
            jamester (1 case(s))
          </option>
        </select>
      </div>
      <div>
        <label
          class="ons-label"
          for="select-to"
        >
          To editor
           
        </label>
        <select
          class="ons-input "
          data-testid="select-to"
          id="select-to"
          name="select"
        >
          <option
            data-testid="select-select-to"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-to-Dave"
            value="Dave"
          >
            Dave (0 case(s))
          </option>
          <option
            data-testid="option-select-to-Jake"
            value="Jake"
          >
            Jake (2 case(s))
          </option>
          <option
            data-testid="option-select-to-Rich"
            value="Rich"
          >
            Rich (0 case(s))
          </option>
        </select>
      </div>
      <div>
        <label
          class="ons-label"
          for="number-of-cases"
        >
          Number of cases
           
        </label>
        <select
          class="ons-input "
          data-testid="number-of-cases"
          id="number-of-cases"
          name="select"
        >
          <option
            data-testid="select-number-of-cases"
            disabled=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-number-of-cases--1"
            selected=""
            value="-1"
          >
            All
          </option>
          <option
            data-testid="option-number-of-cases-50"
            value="50"
          >
            50
          </option>
          <option
            data-testid="option-number-of-cases-45"
            value="45"
          >
            45
          </option>
          <option
            data-testid="option-number-of-cases-40"
            value="40"
          >
            40
          </option>
          <option
            data-testid="option-number-of-cases-35"
            value="35"
          >
            35
          </option>
          <option
            data-testid="option-number-of-cases-30"
            value="30"
          >
            30
          </option>
          <option
            data-testid="option-number-of-cases-25"
            value="25"
          >
            25
          </option>
          <option
            data-testid="option-number-of-cases-20"
            value="20"
          >
            20
          </option>
          <option
            data-testid="option-number-of-cases-15"
            value="15"
          >
            15
          </option>
          <option
            data-testid="option-number-of-cases-10"
            value="10"
          >
            10
          </option>
          <option
            data-testid="option-number-of-cases-5"
            value="5"
          >
            5
          </option>
          <option
            data-testid="option-number-of-cases-1"
            value="1"
          >
            1
          </option>
        </select>
      </div>
      <br />
      <button
        class="ons-btn "
        data-testid="button"
        type="button"
      >
        <span
          class="ons-btn__inner"
        >
          Allocate
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      data-testid="allocation-page-panel"
    >
      <div
        class="ons-panel__body "
      >
        Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor
      </div>
    </div>
    <br />
    <h3>
      FRS2504A
    </h3>
    <div>
      <label
        class="ons-label"
        for="select-from"
      >
        Allocate cases from interviewer
         
      </label>
      <select
        class="ons-input "
        data-testid="select-from"
        id="select-from"
        name="select"
      >
        <option
          data-testid="select-select-from"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-from-bob"
          value="bob"
        >
          bob (2 case(s))
        </option>
        <option
          data-testid="option-select-from-jamester"
          value="jamester"
        >
          jamester (1 case(s))
        </option>
      </select>
    </div>
    <div>
      <label
        class="ons-label"
        for="select-to"
      >
        To editor
         
      </label>
      <select
        class="ons-input "
        data-testid="select-to"
        id="select-to"
        name="select"
      >
        <option
          data-testid="select-select-to"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-to-Dave"
          value="Dave"
        >
          Dave (0 case(s))
        </option>
        <option
          data-testid="option-select-to-Jake"
          value="Jake"
        >
          Jake (2 case(s))
        </option>
        <option
          data-testid="option-select-to-Rich"
          value="Rich"
        >
          Rich (0 case(s))
        </option>
      </select>
    </div>
    <div>
      <label
        class="ons-label"
        for="number-of-cases"
      >
        Number of cases
         
      </label>
      <select
        class="ons-input "
        data-testid="number-of-cases"
        id="number-of-cases"
        name="select"
      >
        <option
          data-testid="select-number-of-cases"
          disabled=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-number-of-cases--1"
          selected=""
          value="-1"
        >
          All
        </option>
        <option
          data-testid="option-number-of-cases-50"
          value="50"
        >
          50
        </option>
        <option
          data-testid="option-number-of-cases-45"
          value="45"
        >
          45
        </option>
        <option
          data-testid="option-number-of-cases-40"
          value="40"
        >
          40
        </option>
        <option
          data-testid="option-number-of-cases-35"
          value="35"
        >
          35
        </option>
        <option
          data-testid="option-number-of-cases-30"
          value="30"
        >
          30
        </option>
        <option
          data-testid="option-number-of-cases-25"
          value="25"
        >
          25
        </option>
        <option
          data-testid="option-number-of-cases-20"
          value="20"
        >
          20
        </option>
        <option
          data-testid="option-number-of-cases-15"
          value="15"
        >
          15
        </option>
        <option
          data-testid="option-number-of-cases-10"
          value="10"
        >
          10
        </option>
        <option
          data-testid="option-number-of-cases-5"
          value="5"
        >
          5
        </option>
        <option
          data-testid="option-number-of-cases-1"
          value="1"
        >
          1
        </option>
      </select>
    </div>
    <br />
    <button
      class="ons-btn "
      data-testid="button"
      type="button"
    >
      <span
        class="ons-btn__inner"
      >
        Allocate
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Given we wish to reallocte cases from an Editor to another Editor should render the allocation page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        data-testid="allocation-page-panel"
      >
        <div
          class="ons-panel__body "
        >
          Reallocate cases from one editor to another editor. All non-completed cases will be transfered
        </div>
      </div>
      <br />
      <h3>
        FRS2504A
      </h3>
      <div>
        <label
          class="ons-label"
          for="select-from"
        >
          Reallocate cases from editor
           
        </label>
        <select
          class="ons-input "
          data-testid="select-from"
          id="select-from"
          name="select"
        >
          <option
            data-testid="select-select-from"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-from-Dave"
            value="Dave"
          >
            Dave (0 case(s))
          </option>
          <option
            data-testid="option-select-from-Jake"
            value="Jake"
          >
            Jake (2 case(s))
          </option>
          <option
            data-testid="option-select-from-Rich"
            value="Rich"
          >
            Rich (0 case(s))
          </option>
        </select>
      </div>
      <div>
        <label
          class="ons-label"
          for="select-to"
        >
          To editor
           
        </label>
        <select
          class="ons-input "
          data-testid="select-to"
          id="select-to"
          name="select"
        >
          <option
            data-testid="select-select-to"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-to-Dave"
            value="Dave"
          >
            Dave (0 case(s))
          </option>
          <option
            data-testid="option-select-to-Jake"
            value="Jake"
          >
            Jake (2 case(s))
          </option>
          <option
            data-testid="option-select-to-Rich"
            value="Rich"
          >
            Rich (0 case(s))
          </option>
        </select>
      </div>
      <div>
        <label
          class="ons-label"
          for="number-of-cases"
        >
          Number of cases
           
        </label>
        <select
          class="ons-input "
          data-testid="number-of-cases"
          id="number-of-cases"
          name="select"
        >
          <option
            data-testid="select-number-of-cases"
            disabled=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-number-of-cases--1"
            selected=""
            value="-1"
          >
            All
          </option>
          <option
            data-testid="option-number-of-cases-50"
            value="50"
          >
            50
          </option>
          <option
            data-testid="option-number-of-cases-45"
            value="45"
          >
            45
          </option>
          <option
            data-testid="option-number-of-cases-40"
            value="40"
          >
            40
          </option>
          <option
            data-testid="option-number-of-cases-35"
            value="35"
          >
            35
          </option>
          <option
            data-testid="option-number-of-cases-30"
            value="30"
          >
            30
          </option>
          <option
            data-testid="option-number-of-cases-25"
            value="25"
          >
            25
          </option>
          <option
            data-testid="option-number-of-cases-20"
            value="20"
          >
            20
          </option>
          <option
            data-testid="option-number-of-cases-15"
            value="15"
          >
            15
          </option>
          <option
            data-testid="option-number-of-cases-10"
            value="10"
          >
            10
          </option>
          <option
            data-testid="option-number-of-cases-5"
            value="5"
          >
            5
          </option>
          <option
            data-testid="option-number-of-cases-1"
            value="1"
          >
            1
          </option>
        </select>
      </div>
      <br />
      <button
        class="ons-btn "
        data-testid="button"
        type="button"
      >
        <span
          class="ons-btn__inner"
        >
          Reallocate
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      data-testid="allocation-page-panel"
    >
      <div
        class="ons-panel__body "
      >
        Reallocate cases from one editor to another editor. All non-completed cases will be transfered
      </div>
    </div>
    <br />
    <h3>
      FRS2504A
    </h3>
    <div>
      <label
        class="ons-label"
        for="select-from"
      >
        Reallocate cases from editor
         
      </label>
      <select
        class="ons-input "
        data-testid="select-from"
        id="select-from"
        name="select"
      >
        <option
          data-testid="select-select-from"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-from-Dave"
          value="Dave"
        >
          Dave (0 case(s))
        </option>
        <option
          data-testid="option-select-from-Jake"
          value="Jake"
        >
          Jake (2 case(s))
        </option>
        <option
          data-testid="option-select-from-Rich"
          value="Rich"
        >
          Rich (0 case(s))
        </option>
      </select>
    </div>
    <div>
      <label
        class="ons-label"
        for="select-to"
      >
        To editor
         
      </label>
      <select
        class="ons-input "
        data-testid="select-to"
        id="select-to"
        name="select"
      >
        <option
          data-testid="select-select-to"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-to-Dave"
          value="Dave"
        >
          Dave (0 case(s))
        </option>
        <option
          data-testid="option-select-to-Jake"
          value="Jake"
        >
          Jake (2 case(s))
        </option>
        <option
          data-testid="option-select-to-Rich"
          value="Rich"
        >
          Rich (0 case(s))
        </option>
      </select>
    </div>
    <div>
      <label
        class="ons-label"
        for="number-of-cases"
      >
        Number of cases
         
      </label>
      <select
        class="ons-input "
        data-testid="number-of-cases"
        id="number-of-cases"
        name="select"
      >
        <option
          data-testid="select-number-of-cases"
          disabled=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-number-of-cases--1"
          selected=""
          value="-1"
        >
          All
        </option>
        <option
          data-testid="option-number-of-cases-50"
          value="50"
        >
          50
        </option>
        <option
          data-testid="option-number-of-cases-45"
          value="45"
        >
          45
        </option>
        <option
          data-testid="option-number-of-cases-40"
          value="40"
        >
          40
        </option>
        <option
          data-testid="option-number-of-cases-35"
          value="35"
        >
          35
        </option>
        <option
          data-testid="option-number-of-cases-30"
          value="30"
        >
          30
        </option>
        <option
          data-testid="option-number-of-cases-25"
          value="25"
        >
          25
        </option>
        <option
          data-testid="option-number-of-cases-20"
          value="20"
        >
          20
        </option>
        <option
          data-testid="option-number-of-cases-15"
          value="15"
        >
          15
        </option>
        <option
          data-testid="option-number-of-cases-10"
          value="10"
        >
          10
        </option>
        <option
          data-testid="option-number-of-cases-5"
          value="5"
        >
          5
        </option>
        <option
          data-testid="option-number-of-cases-1"
          value="1"
        >
          1
        </option>
      </select>
    </div>
    <br />
    <button
      class="ons-btn "
      data-testid="button"
      type="button"
    >
      <span
        class="ons-btn__inner"
      >
        Reallocate
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
